AzureFunction/Utilities.cs

```diff
@@ -0,0 +1,160 @@
+
+    using WebJobsExecutionContext = WebJobs.ExecutionContext;
+
+    public static class Utilities
+    {
+        private const string ImageProvenanceFileName = "ImageProvenance.json";
+        private const string PolicyFileName = "Policies.rego";
+        private const string OutputResultFileName = "Output.txt";
+
+        public static IEnumerable<string> ExecutePolicyCheck(
+            WebJobsExecutionContext executionContext,
+            ILogger log,
+            string imageProvenance,
+            string policy,
+            TaskLogger taskLogger,
+            out string outputLog)
+        {
+            string folderName = string.Format("Policy-{0}", executionContext.InvocationId.ToString("N"));
+            string newFolderPath = Path.Combine(executionContext.FunctionDirectory, folderName);
+
+            log.LogInformation(string.Format("Folder created : {0}", newFolderPath));
+
+            Directory.CreateDirectory(newFolderPath);
+
+            string imageProvenancePath = Path.Combine(newFolderPath, ImageProvenanceFileName);
+            string policyFilePath = Path.Combine(newFolderPath, PolicyFileName);
+
+            string packageName = Regex.Match(policy, @"package\s([a-zA-Z0-9.]+)", RegexOptions.IgnoreCase).Groups[1].Value;
+            log.LogInformation(string.Format("Package name : {0}", packageName));
+
+
+            if (string.IsNullOrWhiteSpace(packageName))
+            {
+                outputLog = "No package name could be inferred from the policy. Cannot continue execution. Ensure that policy contains a package name defined";
+                taskLogger?.Log(outputLog).ConfigureAwait(false);
+
+                return new List<string> { outputLog };
+            }
+
+            string output;
+
+            try
+            {
+                File.WriteAllText(imageProvenancePath, imageProvenance);
+                log.LogInformation("Image provenance file created");
+                File.WriteAllText(policyFilePath, policy);
+                log.LogInformation("Policy content file created");
+
+                string arguments = string.Format(
+                            CultureInfo.InvariantCulture,
+                            "/c \"\"{0}\\opa_windows_amd64.exe\" eval -f pretty --explain notes -i \"{1}\\{2}\" -d \"{1}\\{3}\" \"data.{4}.violations\" > \"{1}\\{5}\"\"",
+                            executionContext.FunctionAppDirectory,
+                            folderName,
+                            ImageProvenanceFileName,
+                            PolicyFileName,
+                            packageName,
+                            OutputResultFileName);
+
+                var process = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = "cmd.exe",
+                        Arguments = arguments,
+                        UseShellExecute = false,
+                        RedirectStandardOutput = true,
```

Comment:
can we RedirectStandardError = true and check process.existcode. if it is not 0 then read StandardError and log errors.

Reply:
I'm handling it in the arguments piping standard error to standard out as well, and reading. For us any result that doesn't make sense in terms of violations is an error, even if exit code is 0